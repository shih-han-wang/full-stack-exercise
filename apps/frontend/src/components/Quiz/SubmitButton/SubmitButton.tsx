import { Button } from '@chakra-ui/react'

type Props = {
  disabled?: boolean
  onClick: () => void
  buttonText?: string
}

const SubmitButton = ({
  disabled = false,
  onClick,
  buttonText = 'Submit'
}: Props) => (
  <Button
    colorScheme='teal'
    disabled={disabled}
    onClick={onClick}
    size='lg'
    type='submit'
    width='320px'
  >
    {buttonText}
  </Button>
)

export default SubmitButton
