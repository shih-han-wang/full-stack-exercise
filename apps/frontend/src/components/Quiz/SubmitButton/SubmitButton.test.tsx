import { render } from '@testing-library/react'
import SubmitButton from './SubmitButton'

const onClickMock = jest.fn()

describe('Components > Quiz > SubmitButton', () => {
  it('should render correctly', async () => {
    const view = render(<SubmitButton onClick={onClickMock} />)

    expect(view.asFragment()).toMatchSnapshot()
  })

  it('should render disabled state correctly', async () => {
    const view = render(<SubmitButton onClick={onClickMock} disabled />)

    expect(view.asFragment()).toMatchSnapshot()
  })

  it('should render buttonTextcorrectly', async () => {
    const view = render(
      <SubmitButton onClick={onClickMock} buttonText='Click me!' />
    )

    expect(view.asFragment()).toMatchSnapshot()
  })
})
