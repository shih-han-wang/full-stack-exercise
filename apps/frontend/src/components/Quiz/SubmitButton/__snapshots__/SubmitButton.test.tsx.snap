// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Quiz > SubmitButton should render buttonTextcorrectly 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-0"
    type="submit"
  >
    Click me!
  </button>
</DocumentFragment>
`;

exports[`Components > Quiz > SubmitButton should render correctly 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-0"
    type="submit"
  >
    Submit
  </button>
</DocumentFragment>
`;

exports[`Components > Quiz > SubmitButton should render disabled state correctly 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-0"
    disabled=""
    type="submit"
  >
    Submit
  </button>
</DocumentFragment>
`;
